### PR TITLE
Add pipeline operators

### PR DESCRIPTION
**Pull request by observant2**
# Description
Adds the pipeline operators `|>` and `<|` known from other functional languages like F# and Elixir.

## Should this change go in the CHANGELOG?

- [x] If this is a fix, user-facing change, a compiler change, or a new paper
      implementation, I have updated [`CHANGELOG_NEXT.md`](https://github.com/idris-lang/Idris2/blob/main/CHANGELOG_NEXT.md) (and potentially also
      `CONTRIBUTORS.md`).

### CODE REVIEW

**Comment by andrevidela** [comment]
Looks like there are some test to fix because now there are two conflicting fixity definitions for `(<|)`

**Comment by observant2** [comment]
Unfortunately I'm stuck here. When I execute "tests/idris2/operators/operators001/Test.idr"
with `%hide Prelude.Ops.infixr.(|>)` I still get the errors:
```
1/1: Building Test (Test.idr)
[-Warning: operator fixity is ambiguous, we are picking Prelude.Ops.infixr.(|>) out of :-]
[- - Prelude.Ops.infixr.(|>), precedence level 9-]
[- - Test.infixr.(|>), precedence level 0-]

[-To remove this warning, use `%hide` with the fixity to remove-]
[-For example: %hide Prelude.Ops.infixr.(|>)-]
```
It suggests me to do what I already did.

**Comment by andrevidela** [comment]
Do the tests run locally? it seems there are still conflicting fixity definitions in the file `Import.idr` for the test `idris2/misc/import009`

**Comment by observant2** [comment]
Now they should pass.

**Comment by observant2** [comment]
Oh, I forgot, someone commented in Discord:
> shouldn't `(|>)` be `infixl` in the prelude?
> also, I think it would make more sense if both pipeline operators had the same precedence

So should I make |> infixl? If they should get the same precedence, which should it be?

**Comment by buzden** [comment]
> So should I make |> infixl?

You definitely should. Even example from your docstring shouldn't typecheck if this operator leaves `infixr`.

> If they should get the same precedence, which should it be?

Precedence should be low enough to be able to form the value you are operating on. Again, in your docstring, you are, actually, applying `|>` after a bunch of `::` (which `[a, b c]` desugars to), so `|>` should have a precedence *lower* than that (and now it's very high). I think that 0 for both is rather a good choice.

Maybe you should add the example from your docstring as a test in order to check that what you intended still works.

**Comment by observant2** [comment]
Okay, that's a good idea, where should this test go? Is there an existing file where it would fit or should I create a new `operators011` folder?

**Comment by buzden** [comment]
> Okay, that's a good idea, where should this test go? Is there an existing file where it would fit or should I create a new `operators011` folder?

It should go here, I think:
https://github.com/idris-lang/Idris2/tree/main/tests/prelude

File: tests/idris2/operators/operators001/Test.idr [hunk 1]
```diff
@@ -6,6 +6,9 @@ import Data.Vect
 private typebind infixr 0 =@
 private infixr 0 -@
 
+%hide Prelude.Ops.infixl.(|>)
+%hide Prelude.Ops.infixr.(|>)
```
Comment (observant2) [inline tests/idris2/operators/operators001/Test.idr:10]: Without this line I would still get warnings, although that operator doesn't exist.

**Review by gallais — CHANGES_REQUESTED** [review]
Please fix the CI errors

**Review by gallais — APPROVED** [review]

**Pull request merged**